### PR TITLE
bump to 0.6.2

### PR DESCRIPTION
**Pull request by ccheetham**
New features
* Azure cloud support

Steeltoe versions:
* upgrade to 2.4.4 from 2.4.3
* add 3.0.0-m3

Documentation
* how-to for adding dependencies

### CODE REVIEW

File: src/Steeltoe.Initializr.WebApp/ClientApp/src/components/BottomLinks.js [hunk 1]
```diff
@@ -6,8 +6,8 @@ export class BottomLinks extends Component {
     return (
 
     <div className="left">
-        <div className="footer">© 2013-2019 Pivotal Software<br/>start.steeltoe.io is powered by<br/>
-            <a href="https://run.pivotal.io/" tabIndex="999">Pivotal Web Services</a><br/>v 0.6.0</div>
+        <div className="footer">© 2013-2020 Pivotal Software<br/>start.steeltoe.io is powered by<br/>
```
Comment (jkonicki) [inline src/Steeltoe.Initializr.WebApp/ClientApp/src/components/BottomLinks.js:9]: This should be VMware Inc. (see steeltoe.io footer).  Also, we can remove the Powered by PWS too.

File: src/Steeltoe.Initializr.WebApp/ClientApp/src/components/Home.js [hunk 2]
```diff
@@ -68,26 +68,21 @@ export class Home extends Component {
...
-        //         "steeltoeVersion": "2.4.0",
-        //         "steeltoeVersionInvalid": "2.4.0 is the lowest version compatible with netcoreapp3.1",
-        //     })
-        // }
-        // else {
-        // ....
-        // }
+        if (name === "steeltoeVersion" && selectedValue === "3.0.0-m3" && this.state.targetFrameworkVersion === "netcoreapp2.1") {
+            this.setState({
+                "steeltoeVersionInvalid": "3.0.0-m3 is not compatible with netcoreapp2.1",
```
Comment (jkonicki) [inline src/Steeltoe.Initializr.WebApp/ClientApp/src/components/Home.js:73]: In this case, where the user selects Steeltoe version 3.0.0, the message should be more like " Steeltoe 3.x.x is only compatible with netcoreapp3.1 and above."

File: src/Steeltoe.Initializr.WebApp/ClientApp/src/components/Home.js [hunk 3]
```diff
@@ -68,26 +68,21 @@ export class Home extends Component {
...
+            })
+        }
+        else if (name === "targetFrameworkVersion" && selectedValue === "netcoreapp2.1" && this.state.steeltoeVersion === "3.0.0-m3") {
+            this.setState({
+                "steeltoeVersionInvalid": "3.0.0-m3 is not compatible with netcoreapp2.1",
```
Comment (jkonicki) [inline src/Steeltoe.Initializr.WebApp/ClientApp/src/components/Home.js:78]: When the user is interacting with the targetFramework version, the message should be specific to what they are changing like "netcoreapp2.1 is only compatible with Steeltoe 2.x.x"

File: test/Steeltoe.Initializr.WebApp.Test/TemplateControllerTests.cs [hunk 4]
```diff
@@ -53,7 +53,7 @@ public TemplateControllerTests(ITestOutputHelper testOutputHelper, TestWebAppFac
         [Fact]
         public async void GetStarterZipTest()
         {
-            var result = await _client.GetAsync("https://localhost/starter.zip?ProjectName=TestCompany.TestProject&Dependencies=Actuator,MySql&Description=Test%20Description&SteeltoeVersion=2.4.3&TemplateVersion=V2&TargetFrameworkVersion=netcoreapp2.1&TemplateShortName=Steeltoe-WebApi");
+            var result = await _client.GetAsync("https://localhost/starter.zip?ProjectName=TestCompany.TestProject&Dependencies=Actuator,MySql&Description=Test%20Description&SteeltoeVersion=2.4.4&TemplateVersion=V2&TargetFrameworkVersion=netcoreapp2.1&TemplateShortName=Steeltoe-WebApi");
```
Comment (jkonicki) [inline test/Steeltoe.Initializr.WebApp.Test/TemplateControllerTests.cs:56]: Can you add a test like this for 3.0.0-m3?

File: test/Steeltoe.Initializr.TemplateEngine.Test/TemplateServiceTests.cs [hunk 5]
```diff
@@ -440,7 +440,7 @@ public async Task CreateTemplate_OauthConnector(ITemplateService templateService
         [ClassData(typeof(AllImplementationsAndTemplates))]
         public async Task CreateTemplate_SqlServer(ITemplateService templateService, string templateName, DotnetTemplateVersion version)
         {
-            var steeltoeVersion = "2.4.3";
+            var steeltoeVersion = "2.4.4";
```
Comment (jkonicki) [inline test/Steeltoe.Initializr.TemplateEngine.Test/TemplateServiceTests.cs:443]: 3.0.0-m3 tests?

File: src/Steeltoe.Initializr.WebApp/ClientApp/src/components/Home.js (hunk 3 above)
Comment (ccheetham) [reply src/Steeltoe.Initializr.WebApp/ClientApp/src/components/Home.js:78]: This was the "path of least resistance" approach.  Ie, it takes advantage of an existing error mechanism that is specific to the Steeltoe version widget so the text was oriented to Steeltoe version. I'll look at alternatives.

Comment (ccheetham) [reply src/Steeltoe.Initializr.WebApp/ClientApp/src/components/Home.js]: Added a new error for the Dotnet target framework widget

File: test/Steeltoe.Initializr.TemplateEngine.Test/TemplateServiceTests.cs (hunk 5 above)
Comment (ccheetham) [reply test/Steeltoe.Initializr.TemplateEngine.Test/TemplateServiceTests.cs:443]: code base will soon be dead

File: test/Steeltoe.Initializr.WebApp.Test/TemplateControllerTests.cs (hunk 4 above)
Comment (ccheetham) [reply test/Steeltoe.Initializr.WebApp.Test/TemplateControllerTests.cs:56]: code base will soon be dead

File: src/templates/Mustache/3.x/Steeltoe-React/mustache.json [hunk 6]
```diff
@@ -172,12 +172,16 @@
...
       "Description": "Target the appropriate Steeltoe version",
       "choices": [
         {
-          "choice": "2.4.3",
-          "Description": "Target Steeltoe version 2.4.3"
+          "choice": "2.4.4",
+          "Description": "Target Steeltoe version 2.4.4"
+        },
+        {
+            "choice": "3.0.0-m2",
```
Comment (jkonicki) [inline src/templates/Mustache/3.x/Steeltoe-React/mustache.json:183]: Should be 3.0.0-m3

File: src/Steeltoe.Initializr.WebApp/ClientApp/src/components/BottomLinks.js [hunk 7]
```diff
@@ -6,8 +6,7 @@ export class BottomLinks extends Component {
     return (
 
     <div className="left">
-        <div className="footer">© 2013-2019 Pivotal Software<br/>start.steeltoe.io is powered by<br/>
-            <a href="https://run.pivotal.io/" tabIndex="999">Pivotal Web Services</a><br/>v 0.6.0</div>
+        <div className="footer">© 2019-2020 VMware, Inc. All Rights Reserved<br/>v 0.6.2</div>
```
Comment (jkonicki) [inline src/Steeltoe.Initializr.WebApp/ClientApp/src/components/BottomLinks.js:9]: Minor - but VMware, Inc. should be a link to https://vmware.com

Comment (ccheetham) [reply src/Steeltoe.Initializr.WebApp/ClientApp/src/components/BottomLinks.js]: added link

File: src/templates/Mustache/3.x/Steeltoe-React/mustache.json (hunk 6 above)
Comment (ccheetham) [reply src/templates/Mustache/3.x/Steeltoe-React/mustache.json]: fixed typo

**Comment by ccheetham** [comment]
re-opening with only 2.x related changes

**Comment by ccheetham** [comment]
#115

**Pull request closed without merging**